modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/prepare/IgniteSqlValidator.java

```diff
@@ -604,6 +609,27 @@ public RelDataType deriveType(SqlValidatorScope scope, SqlNode expr) {
                 });
     }
 
+    /** Check string contain only digit characters.
+     *
+     * @param strVal String representation.
+     * @param pos Current validator position.
+     *
+     * @throws CalciteException if unexpected input string found.
+     **/
+    static void checkStringContainDigitsOnly(@Nullable String strVal, SqlParserPos pos) throws CalciteException {
```

Comment:
CalciteException is a runtime exception, there is no need to declare it.
If this method is only used by the typecoercsion it is better to move it there.

Reply:
got it